faillint/faillint.go

```diff
@@ -11,28 +11,39 @@ import (
 	"golang.org/x/tools/go/analysis"
 )
 
-// Analyzer of the linter
-var Analyzer = &analysis.Analyzer{
-	Name:             "faillint",
-	Doc:              "report unwanted import path usages",
-	Run:              run,
-	RunDespiteErrors: true,
-}
+type faillint struct {
+	paths       string // -paths flag
+	ignoretests bool   // -ignoretests flag
 
-var paths string // -paths flag
+}
 
-func init() {
-	// seems like using init() is the only way to add our own flags
-	Analyzer.Flags.StringVar(&paths, "paths", paths, "import paths to fail")
+// NewAnalyzer create a faillint analyzer
+func NewAnalyzer() *analysis.Analyzer {
+	f := faillint{
+		paths:       "",
+		ignoretests: false,
+	}
+	a := &analysis.Analyzer{
+		Name:             "faillint",
+		Doc:              "report unwanted import path usages",
+		Run:              f.run,
+		RunDespiteErrors: true,
+	}
+	a.Flags.StringVar(&f.paths, "paths", "", "import paths to fail")
+	a.Flags.BoolVar(&f.ignoretests, "ignoretests", false, "ignore all _test.go files and packages.")
+	return a
 }
 
 // Run is the runner for an analysis pass
-func run(pass *analysis.Pass) (interface{}, error) {
-	if paths == "" {
+func (f *faillint) run(pass *analysis.Pass) (interface{}, error) {
+	if f.paths == "" {
+		return nil, nil
+	}
+	if f.ignoretests && strings.Contains(pass.Pkg.Name(), "_test") {
```

Comment:
Hm I wonder if we should we remove this line. We shouldn't look at the package name. I think checking just the files is better to avoid any surprises.

Reply:
Yes I don't think both checks are necessary but the file name check is more important since not all tests are within a test package. I will remove it and push a new version.